packages/bundler-sdk-viem/src/actions.ts

```diff
@@ -892,6 +894,180 @@ export const encodeOperation = (
+      );
+
+      actions.push(
+        {
+          type: "erc20Transfer",
+          args: [
+            srcToken,
+            paraswapAdapter,
+            limitAmount,
+            generalAdapter1,
+            operation.skipRevert,
+          ],
+        },
+        {
+          type: "paraswapBuy",
+          args: [
+            swap.to,
+            swap.data,
+            srcToken,
+            operation.address,
+            swap.offsets,
+            receiver === paraswapAdapter ? generalAdapter1 : receiver,
+            operation.skipRevert,
+          ],
+        },
+        {
+          type: "erc20Transfer",
+          args: [
+            srcToken,
+            generalAdapter1,
+            maxUint256,
+            paraswapAdapter,
+            operation.skipRevert,
+          ],
+        },
+      );
+
+      break;
+    }
+    case "Paraswap_Sell": {
+      if (!("swap" in operation.args))
+        throw new BundlerErrors.MissingSwapData();
+
+      if (paraswapAdapter == null)
+        throw new BundlerErrors.UnexpectedAction("paraswapBuy", chainId);
+
+      const {
+        dstToken,
+        swap,
+        sellEntireBalance = false,
+        receiver,
+      } = operation.args;
+
+      const exactAmountOffset = Number(swap.offsets.exactAmount);
+      const exactAmount = hexToBigInt(
+        slice(swap.data, exactAmountOffset, exactAmountOffset + 32),
+      );
+
+      actions.push(
+        {
+          type: "erc20Transfer",
+          args: [
+            operation.address,
+            paraswapAdapter,
+            sellEntireBalance ? maxUint256 : exactAmount,
+            generalAdapter1,
+            operation.skipRevert,
+          ],
+        },
+        {
+          type: "paraswapSell",
+          args: [
+            swap.to,
+            swap.data,
+            operation.address,
+            dstToken,
+            sellEntireBalance,
+            swap.offsets,
+            receiver === paraswapAdapter ? generalAdapter1 : receiver,
+            operation.skipRevert,
+          ],
+        },
+      );
+
+      if (!sellEntireBalance)
+        actions.push({
+          type: "erc20Transfer",
+          args: [
+            operation.address,
+            generalAdapter1,
+            maxUint256,
+            paraswapAdapter,
+            operation.skipRevert,
+          ],
+        });
+
+      break;
+    }
+    case "Blue_Paraswap_BuyDebt": {
+      if (!("swap" in operation.args))
+        throw new BundlerErrors.MissingSwapData();
+
+      if (paraswapAdapter == null)
+        throw new BundlerErrors.UnexpectedAction("paraswapBuy", chainId);
+
+      const { srcToken, id, swap, onBehalf, receiver } = operation.args;
+
+      const { params } = dataBefore.getMarket(id);
+
+      actions.push(
+        {
+          type: "erc20Transfer",
+          args: [
+            srcToken,
+            paraswapAdapter,
+            maxUint256, // TODO: replace with expected debt including slippage?
```

Comment:
I don't think it's worth it to replace by expected debt since we transfer excess anyways 🤔